enumerables.rb

```diff
@@ -0,0 +1,127 @@
+# rubocop: disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n])
+      n += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n], n)
+      n += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each do |n|
+      result.push(n) if yield(n)
+    end
+    result
+  end
+
+  def my_all?(var = nil)
+    result = true
+    my_each do |n|
+      if block_given?
+        result = false unless yield(n)
+      elsif var.nil?
+        result = false unless n
+      else
+        result = false unless var == n
+      end
+    end
+    result
+  end
```

Comment:
* [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
  ```ruby
    array = [1,2,5,4,7]
    p array.my_all?(Integer) == array.all?(Integer) # false
  ```
* [ ] my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
  ```ruby
    words = %w[dog door rod blade]
    p words.my_all?(/d/) == words.all?(/d/) # false
  ```

Reply:
Accidentally took off equality sign, since Rubocop reported an issue. Now I have disabled that Rubocop property.